Plugins/Flow.Launcher.Plugin.Explorer/Search/WindowsIndex/IndexSearch.cs

```diff
@@ -84,7 +86,8 @@ internal static async Task<List<Result>> ExecuteWindowsIndexSearchAsync(string i
              return results;
         }
 
-        internal async static Task<List<Result>> WindowsIndexSearchAsync(string searchString, string connectionString,
+        internal async static Task<List<Result>> WindowsIndexSearchAsync(string searchString, 
+                                                                  Func<CSearchQueryHelper> queryHelper,
```

Comment:
why not simply accept a CSearchQueryHelper here?

Reply:
Well not much of difference at this stage to pass in the actual object or the function. Just thought to keep it consistent with other parameters, and in the future if we do need to test this function we don't need to mock up the object or change it to a function parameter

Reply:
Hmm even with the delegate seems that we still need to mock the object for testing🧐.

Reply:
Not really, with Explorer's approach you can test each individual function for their behaviour

Reply:
I mean we still need to mock a `CSearchQueryHelper` to test this.

Reply:
It seems that the queryHelper is only used for getting the connectionString, maybe a simple connectionString will be fine here?

Reply:
Creating a connection string calls CSearchManager, that's why I had to change it

Reply:
Sorry i didn't notice the message. 
I think it would be better to keep the dependency cleaner. This method only requires a connection string, so probably creating it outside and pass that in would be better.
I don't think the c search manager belongs to this method, so why not create it outside the method?

Too much delegate may make the code a bit hard to understand.

Reply:
CSearchManager is part of this method call chain, other alternative is to pass in the query helper object, do this instead?

Reply:
Yeah I mean so. I just not quite understand the need to pass a delegate here.

Reply:
done